app/controllers/hyrax/admin/analytics/analytics_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+module Hyrax
+  module Admin
+    module Analytics
+      class AnalyticsController < ApplicationController
+        include Hyrax::SingularSubresourceController
+        before_action :set_months
+        before_action :set_date_range
+        before_action :set_document, only: [:show]
+        with_themed_layout 'dashboard'
+
+        def set_document
+          @document = ::SolrDocument.find(params[:id])
+        end
+
+        def set_months
+          @month_names = 12.downto(1).map { |n| DateTime::ABBR_MONTHNAMES.drop(1)[(Time.zone.today.month - n) % 12] }.reverse
+        end
+
+        def set_date_range
+          @start_date = params[:start_date] || Hyrax.config.analytics_start_date
+          @end_date = params[:end_date] || Time.zone.today + 1.day
+        end
+
+        def date_range
+          "#{@start_date},#{@end_date}"
+        end
+
+        def paginate(results_array, rows: 10)
+          return if results_array.nil?
```

Comment:
does `results_array` sometimes return `nil`? could we make it give an empty array in this case instead?